### PR TITLE
ENT-10921: Improved failure logging during federated reporting schema import

### PR DESCRIPTION
**Pull request by craigcomstock**
Saves full log in /var/cfengine/outputs

TODO: clean up old /var/cfengine/outputs/*-schema-attach-*.log files

Ticket: ENT-10789
Changelog: title

### CODE REVIEW

**Comment by craigcomstock** [comment]
Should wait for affected users to report that this change is helpful in debugging.

**Review by nickanderson — APPROVED** [review]

File: templates/federated_reporting/import.sh [hunk 1]
```diff
@@ -124,9 +125,17 @@ log "Attaching schemas"
 for file in $dump_files; do
   if [ ! -f "${file}.failed" ]; then
     hostkey=$(basename "$file" | cut -d. -f1)
+    logfile="$WORKDIR"/outputs/"$hostkey"-schema-attach-$(date +%F-%T)-failure.log
     "$CFE_BIN_DIR"/psql -U $CFE_FR_DB_USER -d cfdb --set "ON_ERROR_STOP=1" \
                         -c "SET SCHEMA 'public'; SELECT attach_feeder_schema('$hostkey', ARRAY[$table_whitelist]);" \
```
Comment (craigcomstock) [inline templates/federated_reporting/import.sh:130]: @nickanderson how would you integrate/configure debug logging for federated imports?

We can add
```
                        -c "SET client_min_messages TO DEBUG5" \
```

In this line and any other psql commands we run here and capture logs as I have done here to track down problems easier.z 

Maybe an augment to set the log level or just "debug" as a boolean would be easier.

Comment (nickanderson) [reply templates/federated_reporting/import.sh:130]: First thing that comes to mind would be a new variable, defaulted to NOTICE but overridable via augments.

Then someone could set that variable to any of the allowed values:

DEBUG5, DEBUG4, DEBUG3, DEBUG2, DEBUG1, LOG, NOTICE, WARNING, and ERROR.

Comment (nickanderson) [reply templates/federated_reporting/import.sh:130]: Oh, that's a shell script, not a renderd mustache file.

Comment (craigcomstock) [reply templates/federated_reporting/import.sh:135]: my proposal here doesn't work yet... bash quoting :'(

Comment (craigcomstock) [reply templates/federated_reporting/import.sh]: I tested what I have here now and think this looks good. I bailed on trying to figure out bash escaping, not worth the effort so just copy/pasted the command twice.

**Review by nickanderson — APPROVED** [review]

**Comment by craigcomstock** [comment]
cherry picks
https://github.com/cfengine/masterfiles/pull/2781
https://github.com/cfengine/masterfiles/pull/2780

**Pull request merged**